### ISSUE TITLE
Angular Language Service 0.901 missing symbols from core.d.ts

### ISSUE THREAD

**Issue by kingpric** [angular/vscode-ng-language-service#713]
After updating to version 0.901.0, extension not activating on VS code 1.43.2. Its not showing on Developer Running Extensions. But no issue with 0.900.18.

**Comment by ayazhafiz** [comment]
Can you provide some more information. Is there a log on the output of View > Toggle Output > Angular Language Service? Otherwise, what does your project look like, can you confirm that you have an angular.json or `@angular/core` installed in your node_modules?

**Comment by kingpric** [comment]
There is no log for Angular Language Service. Also @angular/core:9.1.0 present on node_modules 
Here is snippet of angular.json

`{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "ng2020": {
      "projectType": "application",
      "schematics": {},
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/ng2020",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": "src/polyfills.ts",
            "tsConfig": "tsconfig.app.json",
            "aot": true,
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "src/styles.css"
            ],
            "scripts": []
          },
          "configurations": {
            "production": {
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "extractCss": true,
              "namedChunks": false,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true,
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "2mb",
                  "maximumError": "5mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "6kb",
                  "maximumError": "10kb"
                }
              ]
            }
          }
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "options": {
            "browserTarget": "ng2020:build"
          },
          "configurations": {
            "production": {
              "browserTarget": "ng2020:build:production"
            }
          }
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "ng2020:build"
          }
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "main": "src/test.ts",
            "polyfills": "src/polyfills.ts",
            "tsConfig": "tsconfig.spec.json",
            "karmaConfig": "karma.conf.js",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "src/styles.css"
            ],
            "scripts": []
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "tsconfig.app.json",
              "tsconfig.spec.json",
              "e2e/tsconfig.json"
            ],
            "exclude": [
              "**/node_modules/**"
            ]
          }
        },
        "e2e": {
          "builder": "@angular-devkit/build-angular:protractor",
          "options": {
            "protractorConfig": "e2e/protractor.conf.js",
            "devServerTarget": "ng2020:serve"
          },
          "configurations": {
            "production": {
              "devServerTarget": "ng2020:serve:production"
            }
          }
        }
      }
    }},
  "defaultProject": "ng2020"
}
`

**Comment by ayazhafiz** [comment]
Any way you can provide an example project to reproduce this?

**Comment by kingpric** [comment]
Issue can be replicated on using VS Code version 1.43.2 and creating new angular project, also installing extension angular.ng-template version 0.901.0

But on downgrading angular.ng-template  to 0.900.18 extension working as expected, Angular Language Service log showing on output, also it can be seen on Developer:Show Running Extension.

**Comment by alexermolov** [comment]
After update for typescript and language service to the latest versions no more crashes but extension is still not working. No more navigation from templates to source code

**Comment by ayazhafiz** [comment]
sorry for the delay. i will take a look at this today.

**Comment by ayazhafiz** [comment]
Hmm.. just tried this with a new Angular app on VSCode 1.43.2, with 0.901.0 of the extension. The language service started up and seems to be working fine.

Could anyone experiencing this issue provide a link to repository for which they experience this issue? What platform (windows, linux, mac) are you using? What package manager (yarn, npm, etc) are you using to install npm dependencies?

**Comment by alexermolov** [comment]
actually I've tried to reproduce this problem on small and medium projects and didn't notice that problem but on real enterprise app it starts to happen (about 1000 components wrapped with modules)

![record](https://user-images.githubusercontent.com/25246926/78901906-0f5c3c80-7a8a-11ea-8dd9-9d4ceb1325a2.gif)

**Comment by ayazhafiz** [comment]
Our startup time isn't great -- it's been the cause of a number of perf reports. Maybe this is a manifestation of that, wherein the extension takes forever to load up and VSCode terms it.

@kyliau do you have any insights?

**Comment by tonivj5** [comment]
Hi all, I have been experiencing this issue for a long time ago. With the update to > 9.0.0, mostly, the plugin has stopped working. I have noticed that it always breaks when I use `Go to reference`, or add a new module at my `SharedModule` or something similar, and once it breaks, when I go to another component were it was working, it shows these fails in the template:

```
Property binding ngIf not used by any directive on an embedded template. Make sure that the property name is spelled correctly and all directives are listed in the "@NgModule.declarations".

Property binding ngForOf not used by any directive on an embedded template. Make sure that the property name is spelled correctly and all directives are listed in the "@NgModule.declarations".

Can't bind to 'ngForOf' since it isn't a known property of 'span'.

The pipe 'async' could not be found
```

If I edited the template and save it, the errors go away and the plugin doesn't work anymore (it doesn't show any error neither give me any intellisense) until I restart the vscode. Practically, I'm developing without any help (at template)

I have tried of reproduce it, but it is complicated. On a new project (`ng new`) I was able of do it, but only once time :confused: I don't know exactly what are the steps to break it, but in a large project it's easier. How can I show debug information to see where it breaks? I can't share my project because it's closed :sweat:

PD: I have found time today to report it, and I have seen this issue, so before I open a new issue I think I should comment it here.

**Comment by ayazhafiz** [comment]
Yeah I’m inclined to believe this is the language service’s poor startup time getting worse because of large products. VSCode terminates language service requests if they take too long to respond. It’s also possible that resolution of directives and pipes just isn’t happening correctly inside the Angular compiler. @kyliau are you aware of any changes to Angular in 9.1 that may be related to this?

**Comment by ayazhafiz** [comment]
> How can I show debug information to see where it breaks? I can't share my project because it's closed 😓

@tonivj5 if you are willing, you could share the debug logs corresponding to the time you see the extension stop working. The log file path is listed in the extension output (View > Output > Angular Language Service)

**Comment by tonivj5** [comment]
Sure!

@ayazhafiz, here is my log after it breaks:
```log
Info 0    [9:56:42.707] Format host information updated
Info 1    [9:56:42.707] reload projects.
Info 2    [9:56:42.707] Structure before ensureProjectForOpenFiles:
Info 3    [9:56:42.707] Structure after ensureProjectForOpenFiles:
Info 4    [9:56:42.707] Host file extension mappings updated
Info 5    [9:56:42.712] Search path: /home/my/project/src/app/theme/components/list-box
Info 6    [9:56:42.713] For info: /home/my/project/src/app/theme/components/list-box/list.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 7    [9:56:42.714] Opened configuration file /home/my/project/tsconfig.json
Info 8    [9:56:42.795] Config: /home/my/project/tsconfig.json : {
 "rootNames": [
  "/home/my/project/jest.setup.ts",
  "/home/my/project/e2e/src/app.e2e-spec.ts",
  "/home/my/project/e2e/src/app.po.ts",
  "/home/my/project/src/main.ts",
  "/home/my/project/src/polyfills.ts",
  "/home/my/project/src/typings.d.ts",
  "/home/my/project/src/zone-flags.ts",
  "/home/my/project/src/app/app-routing.module.ts",
  "/home/my/project/src/app/app.component.ts",
  "/home/my/project/src/app/app.module.ts",
  "/home/my/project/src/app/core/core.module.ts",
  "/home/my/project/src/app/core/module-import-guard.ts",
  "/home/my/project/src/app/core/shared.module.ts",
  "/home/my/project/src/app/core/guards/auth.guard.service.ts",
  "/home/my/project/src/app/core/interceptors/index.ts",
  "/home/my/project/src/app/core/interceptors/jwt.interceptor.ts",
  "/home/my/project/src/app/core/interceptors/logout-on-401.interceptor.ts",
  "/home/my/project/src/app/core/interceptors/notify-on-500.interceptor.ts",
  "/home/my/project/src/app/core/interceptors/notify-on-timeout.interceptor.ts",
  "/home/my/project/src/app/core/interceptors/simplify-http-error.interceptor.ts",
  -----_____----- a lot of more files (~420 files) -------______----
  "/home/my/project/src/app/theme/components/table/table-sortable-header.directive.ts",
  "/home/my/project/src/app/theme/components/table/table.component.spec.ts",
  "/home/my/project/src/app/theme/components/table/table.component.ts",
  "/home/my/project/src/app/theme/components/table/table.module.ts",
  "/home/my/project/src/app/theme/components/table/table.ts",
  "/home/my/project/src/app/theme/components/table/custom-cells/component-cell.component.ts",
  "/home/my/project/src/app/theme/components/table/custom-cells/composition-cell.component.ts",
  "/home/my/project/src/app/theme/components/table/custom-cells/index.ts",
  "/home/my/project/src/app/theme/components/table/custom-cells/template-cell.component.ts",
  "/home/my/project/src/app/theme/components/table/tokens/context.token.ts",
  "/home/my/project/src/app/theme/components/table/tokens/index.ts",
  "/home/my/project/src/app/theme/components/table/tokens/item.token.ts",
  "/home/my/project/src/app/theme/components/table/tokens/property.token.ts",
  "/home/my/project/src/app/theme/components/table/tokens/value.token.ts",
  "/home/my/project/src/app/theme/directives/clearlable.directive.ts",
  "/home/my/project/src/app/theme/directives/error.directive.ts",
  "/home/my/project/src/app/theme/directives/errors.directive.ts",
  "/home/my/project/src/app/theme/directives/form-control-valid.directive.ts",
  "/home/my/project/src/app/theme/directives/has-permission.directive.ts",
  "/home/my/project/src/app/theme/directives/index.ts",
  "/home/my/project/src/app/theme/directives/ng-const.directive.ts",
  "/home/my/project/src/app/theme/directives/ng-let.directive.ts",
  "/home/my/project/src/app/theme/directives/ng-var.directive.ts",
  "/home/my/project/src/app/theme/directives/tooltip.directive.ts",
  "/home/my/project/src/app/theme/layout/layout.component.ts",
  "/home/my/project/src/app/theme/pipes/capitalize.pipe.ts",
  "/home/my/project/src/app/theme/pipes/date-ago.pipe.ts",
  "/home/my/project/src/app/theme/pipes/entries.pipe.ts",
  "/home/my/project/src/app/theme/pipes/has-permission.pipe.ts",
  "/home/my/project/src/app/theme/pipes/index.ts",
  "/home/my/project/src/app/theme/pipes/memoize.pipe.ts",
  "/home/my/project/src/app/theme/pipes/money.pipe.ts",
  "/home/my/project/src/app/theme/pipes/number-with-commas.pipe.ts",
  "/home/my/project/src/app/theme/pipes/pair.pipe.ts",
  "/home/my/project/src/app/theme/pipes/plural.pipe.ts",
  "/home/my/project/src/app/theme/pipes/round.pipe.ts",
  "/home/my/project/src/app/theme/pipes/split.pipe.ts",
  "/home/my/project/src/app/theme/pipes/timing.pipe.ts",
  "/home/my/project/src/app/theme/pipes/to$.ts",
  "/home/my/project/src/app/theme/pipes/track-by-property.pipe.ts",
  "/home/my/project/src/app/theme/pipes/values.pipe.ts",
  "/home/my/project/src/app/theme/styles/theme.default.ts",
  "/home/my/project/src/assets/i18n/de.ts",
  "/home/my/project/src/assets/i18n/en.ts",
  "/home/my/project/src/environments/environment.common.ts",
  "/home/my/project/src/environments/environment.prod.ts",
  "/home/my/project/src/environments/environment.staging.ts",
  "/home/my/project/src/environments/environment.ts",
  "/home/my/project/src/environments/environment.type.ts"
 ],
 "options": {
  "module": 99,
  "baseUrl": "/home/my/project/src",
  "outDir": "/home/my/project/dist/out-tsc",
  "sourceMap": true,
  "declaration": false,
  "importHelpers": true,
  "moduleResolution": 2,
  "downlevelIteration": true,
  "emitDecoratorMetadata": true,
  "experimentalDecorators": true,
  "target": 2,
  "typeRoots": [
   "/home/my/project/node_modules/@types"
  ],
  "lib": [
   "lib.es2018.d.ts",
   "lib.dom.d.ts",
   "lib.es2019.array.d.ts",
   "lib.es2019.object.d.ts"
  ],
  "paths": {
   "@state/*": [
    "app/core/state/*"
   ],
   "@core/*": [
    "app/core/*"
   ],
   "@assets/*": [
    "assets/*"
   ],
   "@pages/*": [
    "app/pages/*"
   ],
   "@theme/*": [
    "app/theme/*"
   ]
  },
  "configFilePath": "/home/my/project/tsconfig.json"
 }
}
Info 9    [9:56:44.794] Loading global plugin @angular/language-service
Info 10   [9:56:44.794] Enabling plugin @angular/language-service from candidate paths: /home/my/.vscode/extensions/angular.ng-template-0.901.0/server/node_modules/@angular/language-service/bundles/language-service.umd.js,/home/my/project/node_modules/typescript/lib/tsserverlibrary.js/../../..
Info 11   [9:56:44.794] Loading @angular/language-service from /home/my/.vscode/extensions/angular.ng-template-0.901.0/server/node_modules/@angular/language-service/bundles/language-service.umd.js (resolved to /home/my/.vscode/extensions/angular.ng-template-0.901.0/server/node_modules/@angular/language-service/bundles/language-service.umd.js/node_modules)
Info 12   [9:56:44.849] Plugin validation succeded
Info 13   [9:56:44.919] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 14   [9:56:49.301] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 1 structureChanged: true Elapsed: 4382ms
Info 15   [9:56:49.302] Project '/home/my/project/tsconfig.json' (Configured) 
Info 16   [9:56:49.302]         Files (1611)

Info 17   [9:56:49.302] -----------------------------------------------
Info 18   [9:56:49.311] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 19   [9:56:49.351] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 2 structureChanged: false Elapsed: 40ms
Info 20   [9:56:49.351] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 21   [9:56:49.358] Search path: /home/my/project/src/app/theme/components/list-box
Info 22   [9:56:49.358] For info: /home/my/project/src/app/theme/components/list-box/list-box.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 23   [10:3:3.276] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 24   [10:3:3.277] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 25   [10:3:3.549] Search path: /home/my/project/src/app/theme/components/list-box
Info 26   [10:3:3.549] For info: /home/my/project/src/app/theme/components/list-box/infinite-scroll.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 27   [10:3:3.748] Search path: /home/my/project/src/app/theme/components/list-box
Info 28   [10:3:3.748] For info: /home/my/project/src/app/theme/components/list-box/infinite-scroll.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 29   [10:3:4.484] Search path: /home/my/project/src/app/theme/components/list-box
Info 30   [10:3:4.484] For info: /home/my/project/src/app/theme/components/list-box/infinite-scroll.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 31   [10:3:23.428] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop
Info 32   [10:3:23.429] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/drag-utils.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 33   [10:3:34.571] Search path: /home/my/project/src/app/pages/roles/role-form/inputs
Info 34   [10:3:34.571] For info: /home/my/project/src/app/pages/roles/role-form/inputs/list-box-input.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 35   [10:3:37.687] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 36   [10:3:37.760] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 3 structureChanged: false Elapsed: 73ms
Info 37   [10:3:37.760] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 38   [10:3:38.370] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 39   [10:3:38.424] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 4 structureChanged: false Elapsed: 53ms
Info 40   [10:3:38.424] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 41   [10:3:41.204] Search path: /home/my/project/node_modules/@angular/common
Info 42   [10:3:41.205] For info: /home/my/project/node_modules/@angular/common/common.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 43   [10:3:41.205] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 44   [10:3:41.442] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 5 structureChanged: true Elapsed: 237ms
Info 45   [10:3:41.442] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 46   [10:3:41.447] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 47   [10:3:41.621] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 6 structureChanged: true Elapsed: 174ms
Info 48   [10:3:41.621] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 49   [10:3:41.889] Search path: /home/my/project/node_modules/@angular/common
Info 50   [10:3:41.889] For info: /home/my/project/node_modules/@angular/common/common.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 51   [10:3:41.889] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 52   [10:3:42.64] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 7 structureChanged: true Elapsed: 175ms
Info 53   [10:3:42.64] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 54   [10:3:42.267] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 55   [10:3:42.460] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 8 structureChanged: true Elapsed: 193ms
Info 56   [10:3:42.461] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 57   [10:3:50.307] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 58   [10:3:50.383] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 9 structureChanged: false Elapsed: 76ms
Info 59   [10:3:50.383] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 60   [10:3:53.642] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 61   [10:3:53.718] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 10 structureChanged: false Elapsed: 76ms
Info 62   [10:3:53.718] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 63   [10:3:54.658] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 64   [10:3:54.725] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 11 structureChanged: false Elapsed: 66ms
Info 65   [10:3:54.725] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 66   [10:3:55.567] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 67   [10:3:55.644] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 12 structureChanged: false Elapsed: 76ms
Info 68   [10:3:55.644] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 69   [10:3:56.493] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 70   [10:3:56.562] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 13 structureChanged: false Elapsed: 68ms
Info 71   [10:3:56.562] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 72   [10:3:57.224] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 73   [10:3:57.295] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 14 structureChanged: false Elapsed: 71ms
Info 74   [10:3:57.295] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 75   [10:3:58.802] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 76   [10:3:58.875] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 15 structureChanged: false Elapsed: 73ms
Info 77   [10:3:58.875] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 78   [10:4:2.981] Search path: /home/my/project/node_modules/@angular/common
Info 79   [10:4:2.981] For info: /home/my/project/node_modules/@angular/common/common.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 80   [10:4:2.981] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 81   [10:4:3.160] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 16 structureChanged: true Elapsed: 179ms
Info 82   [10:4:3.161] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 83   [10:4:3.167] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 84   [10:4:3.343] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 17 structureChanged: true Elapsed: 176ms
Info 85   [10:4:3.343] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 86   [10:4:3.546] Search path: /home/my/project/node_modules/@angular/common
Info 87   [10:4:3.546] For info: /home/my/project/node_modules/@angular/common/common.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 88   [10:4:3.546] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 89   [10:4:3.736] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 18 structureChanged: true Elapsed: 190ms
Info 90   [10:4:3.736] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 91   [10:4:3.881] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 92   [10:4:4.51] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 19 structureChanged: true Elapsed: 170ms
Info 93   [10:4:4.51] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 94   [10:4:4.579] Search path: /home/my/project/node_modules/@angular/common
Info 95   [10:4:4.579] For info: /home/my/project/node_modules/@angular/common/common.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 96   [10:4:4.579] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 97   [10:4:4.822] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 20 structureChanged: true Elapsed: 243ms
Info 98   [10:4:4.822] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 99   [10:4:6.482] Search path: /home/my/project/node_modules/typescript/lib
Info 100  [10:4:6.482] For info: /home/my/project/node_modules/typescript/lib/lib.es5.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 101  [10:4:6.482] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 102  [10:4:6.559] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 21 structureChanged: false Elapsed: 77ms
Info 103  [10:4:6.559] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 104  [10:4:6.561] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 105  [10:4:6.621] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 22 structureChanged: false Elapsed: 60ms
Info 106  [10:4:6.621] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 107  [10:4:8.215] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop
Info 108  [10:4:8.215] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/drag-events.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 109  [10:4:8.466] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop
Info 110  [10:4:8.466] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/drag-events.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 111  [10:4:11.890] Search path: /home/my/project/src/app/theme/components/list-box
Info 112  [10:4:11.891] For info: /home/my/project/src/app/theme/components/list-box/infinite-scroll.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 113  [10:4:13.45] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 114  [10:4:13.46] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 115  [10:4:25.583] Search path: /home/my/project/src/app/theme/components/list-box
Info 116  [10:4:25.583] For info: /home/my/project/src/app/theme/components/list-box/infinite-scroll.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 117  [10:4:25.583] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 118  [10:4:25.797] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 23 structureChanged: true Elapsed: 214ms
Info 119  [10:4:25.797] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 120  [10:4:36.794] Search path: /home/my/project/src/app/pages
Info 121  [10:4:36.794] For info: /home/my/project/src/app/pages/pages.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 122  [10:4:40.400] Search path: /home/my/project/src/app/pages
Info 123  [10:4:40.401] For info: /home/my/project/src/app/pages/pages.module.ts :: Config file name: /home/my/project/tsconfig.json
Info 124  [10:4:46.148] Search path: /home/my/project/src/app/pages/roles
Info 125  [10:4:46.149] For info: /home/my/project/src/app/pages/roles/roles.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 126  [10:4:54.561] Search path: /home/my/project/src/app
Info 127  [10:4:54.562] For info: /home/my/project/src/app/app.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 128  [10:5:57.541] Search path: /home/my/project/src/app/theme/components/list-box
Info 129  [10:5:57.541] For info: /home/my/project/src/app/theme/components/list-box/infinite-scroll.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 130  [10:6:22.563] Search path: /home/my/project/src/app/theme/components/login
Info 131  [10:6:22.563] For info: /home/my/project/src/app/theme/components/login/login.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 132  [10:6:24.189] Search path: /home/my/project/src/app/theme/components/list-box
Info 133  [10:6:24.189] For info: /home/my/project/src/app/theme/components/list-box/infinite-scroll.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 134  [10:6:29.254] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 135  [10:6:29.254] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drag.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 136  [10:6:29.736] Search path: /home/my/project/node_modules/@angular/common
Info 137  [10:6:29.736] For info: /home/my/project/node_modules/@angular/common/common.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 138  [10:6:29.736] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 139  [10:6:29.912] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 24 structureChanged: true Elapsed: 176ms
Info 140  [10:6:29.912] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 141  [10:6:29.916] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 142  [10:6:30.78] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 25 structureChanged: true Elapsed: 162ms
Info 143  [10:6:30.78] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 144  [10:6:30.354] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 145  [10:6:30.354] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 146  [10:6:30.676] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 147  [10:6:30.676] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 148  [10:6:30.815] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 149  [10:6:30.816] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 150  [10:6:31.9] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 151  [10:6:31.10] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 152  [10:6:31.296] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 153  [10:6:31.296] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 154  [10:6:31.554] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 155  [10:6:31.554] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 156  [10:6:31.585] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 157  [10:6:31.586] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 158  [10:6:31.881] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 159  [10:6:31.881] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 160  [10:6:32.10] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 161  [10:6:32.12] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 162  [10:6:32.121] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 163  [10:6:32.122] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 164  [10:6:36.676] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 165  [10:6:36.676] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 166  [10:6:37.648] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 167  [10:6:37.649] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 168  [10:6:37.845] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 169  [10:6:37.846] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 170  [10:6:37.941] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 171  [10:6:37.942] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 172  [10:6:38.320] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 173  [10:6:38.321] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 174  [10:6:40.355] Search path: /home/my/project/src/app/theme/components/list-box
Info 175  [10:6:40.356] For info: /home/my/project/src/app/theme/components/list-box/infinite-scroll.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 176  [10:6:40.729] Search path: /home/my/project/src/app/theme/components/list-box
Info 177  [10:6:40.729] For info: /home/my/project/src/app/theme/components/list-box/infinite-scroll.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 178  [10:6:40.757] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 179  [10:6:40.758] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 180  [10:6:45.608] Search path: /home/my/project/node_modules/@angular/common
Info 181  [10:6:45.608] For info: /home/my/project/node_modules/@angular/common/common.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 182  [10:6:45.608] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 183  [10:6:45.770] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 26 structureChanged: true Elapsed: 162ms
Info 184  [10:6:45.770] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 185  [10:6:45.775] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 186  [10:6:45.923] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 27 structureChanged: true Elapsed: 148ms
Info 187  [10:6:45.924] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 188  [10:6:46.81] Search path: /home/my/project/node_modules/@angular/common
Info 189  [10:6:46.82] For info: /home/my/project/node_modules/@angular/common/common.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 190  [10:6:46.82] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 191  [10:6:46.233] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 28 structureChanged: true Elapsed: 151ms
Info 192  [10:6:46.233] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 193  [10:6:46.236] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 194  [10:6:46.391] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 29 structureChanged: true Elapsed: 155ms
Info 195  [10:6:46.391] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 196  [10:6:46.525] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 197  [10:6:46.525] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drag.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 198  [10:6:47.93] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 199  [10:6:47.93] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drag.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 200  [10:6:50.13] Search path: /home/my/project/node_modules/@angular/common
Info 201  [10:6:50.13] For info: /home/my/project/node_modules/@angular/common/common.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 202  [10:6:50.13] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 203  [10:6:50.164] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 30 structureChanged: true Elapsed: 151ms
Info 204  [10:6:50.165] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 205  [10:6:50.167] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 206  [10:6:50.309] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 31 structureChanged: true Elapsed: 142ms
Info 207  [10:6:50.309] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 208  [12:6:26.85] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 209  [12:6:26.85] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 210  [12:6:28.939] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 211  [12:6:29.13] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 32 structureChanged: false Elapsed: 74ms
Info 212  [12:6:29.13] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 213  [12:6:30.57] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 214  [12:6:30.120] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 33 structureChanged: false Elapsed: 63ms
Info 215  [12:6:30.120] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 216  [12:6:30.396] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 217  [12:6:30.458] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 34 structureChanged: false Elapsed: 62ms
Info 218  [12:6:30.458] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 219  [12:6:31.339] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 220  [12:6:31.429] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 35 structureChanged: false Elapsed: 90ms
Info 221  [12:6:31.429] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 222  [12:6:34.331] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 223  [12:6:34.383] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 36 structureChanged: false Elapsed: 52ms
Info 224  [12:6:34.383] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 225  [12:6:36.17] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 226  [12:6:36.82] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 37 structureChanged: false Elapsed: 65ms
Info 227  [12:6:36.82] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 228  [12:6:37.248] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 229  [12:6:37.318] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 38 structureChanged: false Elapsed: 70ms
Info 230  [12:6:37.318] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 231  [12:6:38.178] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 232  [12:6:38.239] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 39 structureChanged: false Elapsed: 61ms
Info 233  [12:6:38.239] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 234  [12:6:52.188] Search path: /home/my/project/src/app/pages/roles/role-form/inputs
Info 235  [12:6:52.189] For info: /home/my/project/src/app/pages/roles/role-form/inputs/permission.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 236  [12:6:52.702] Search path: /home/my/project/src/app/pages/roles/role-form/inputs
Info 237  [12:6:52.702] For info: /home/my/project/src/app/pages/roles/role-form/inputs/permission.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 238  [12:6:56.313] Search path: /home/my/project/src/app/core/models
Info 239  [12:6:56.313] For info: /home/my/project/src/app/core/models/permission.ts :: Config file name: /home/my/project/tsconfig.json
Info 240  [12:7:0.783] Search path: /home/my/project/node_modules/@angular/forms
Info 241  [12:7:0.783] For info: /home/my/project/node_modules/@angular/forms/forms.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 242  [12:7:0.783] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 243  [12:7:0.846] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 40 structureChanged: false Elapsed: 63ms
Info 244  [12:7:0.846] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 245  [12:7:0.848] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 246  [12:7:0.899] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 41 structureChanged: false Elapsed: 51ms
Info 247  [12:7:0.899] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 248  [12:7:1.374] Search path: /home/my/project/node_modules/@angular/common
Info 249  [12:7:1.375] For info: /home/my/project/node_modules/@angular/common/common.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 250  [12:7:1.375] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 251  [12:7:1.544] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 42 structureChanged: true Elapsed: 169ms
Info 252  [12:7:1.544] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 253  [12:7:1.748] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 254  [12:7:1.942] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 43 structureChanged: true Elapsed: 194ms
Info 255  [12:7:1.942] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 256  [12:7:2.279] Search path: /home/my/project/node_modules/@angular/common
Info 257  [12:7:2.280] For info: /home/my/project/node_modules/@angular/common/common.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 258  [12:7:2.280] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 259  [12:7:2.475] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 44 structureChanged: true Elapsed: 195ms
Info 260  [12:7:2.475] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 261  [12:7:2.481] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 262  [12:7:2.632] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 45 structureChanged: true Elapsed: 151ms
Info 263  [12:7:2.632] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 264  [12:7:4.353] Search path: /home/my/project/node_modules/@angular/core
Info 265  [12:7:4.353] For info: /home/my/project/node_modules/@angular/core/core.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 266  [12:7:4.353] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 267  [12:7:4.599] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 46 structureChanged: true Elapsed: 246ms
Info 268  [12:7:4.599] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 269  [12:7:4.606] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 270  [12:7:4.812] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 47 structureChanged: true Elapsed: 206ms
Info 271  [12:7:4.812] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 272  [12:7:4.965] Search path: /home/my/project/src/app/theme/components/list-box
Info 273  [12:7:4.965] For info: /home/my/project/src/app/theme/components/list-box/infinite-scroll.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 274  [12:7:5.305] Search path: /home/my/project/src/app/theme/components/list-box
Info 275  [12:7:5.305] For info: /home/my/project/src/app/theme/components/list-box/infinite-scroll.component.ts :: Config file name: /home/my/project/tsconfig.json
Info 276  [12:7:5.500] Search path: /home/my/project/node_modules/@angular/core
Info 277  [12:7:5.500] For info: /home/my/project/node_modules/@angular/core/core.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 278  [12:7:5.501] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 279  [12:7:5.738] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 48 structureChanged: true Elapsed: 237ms
Info 280  [12:7:5.738] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 281  [12:7:5.745] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 282  [12:7:5.958] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 49 structureChanged: true Elapsed: 213ms
Info 283  [12:7:5.958] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 284  [12:7:18.161] Search path: /home/my/project/node_modules/@angular/core
Info 285  [12:7:18.161] For info: /home/my/project/node_modules/@angular/core/core.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 286  [12:7:18.161] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 287  [12:7:18.403] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 50 structureChanged: true Elapsed: 241ms
Info 288  [12:7:18.403] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 289  [12:7:18.410] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 290  [12:7:18.627] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 51 structureChanged: true Elapsed: 217ms
Info 291  [12:7:18.627] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 292  [12:7:27.572] Search path: /home/my/project/node_modules/@angular/common
Info 293  [12:7:27.573] For info: /home/my/project/node_modules/@angular/common/common.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 294  [12:7:27.573] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 295  [12:7:27.721] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 52 structureChanged: true Elapsed: 148ms
Info 296  [12:7:27.721] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 297  [12:7:27.724] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 298  [12:7:27.876] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 53 structureChanged: true Elapsed: 152ms
Info 299  [12:7:27.876] Different program with same set of files:: oldProgram.structureIsReused:: 0
Info 300  [12:7:38.744] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 301  [12:7:38.744] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 302  [12:7:38.782] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 303  [12:7:38.783] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 304  [12:7:39.19] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 305  [12:7:39.20] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 306  [12:7:40.86] Search path: /home/my/project/node_modules/@angular/cdk/drag-drop/directives
Info 307  [12:7:40.88] For info: /home/my/project/node_modules/@angular/cdk/drag-drop/directives/drop-list.d.ts :: Config file name: /home/my/project/tsconfig.json
Info 308  [12:7:44.256] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 309  [12:7:44.320] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 54 structureChanged: false Elapsed: 64ms
Info 310  [12:7:44.320] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 311  [12:7:45.423] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 312  [12:7:45.500] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 55 structureChanged: false Elapsed: 77ms
Info 313  [12:7:45.500] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 314  [12:7:46.554] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 315  [12:7:46.600] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 56 structureChanged: false Elapsed: 46ms
Info 316  [12:7:46.600] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 317  [12:8:10.548] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 318  [12:8:10.630] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 57 structureChanged: false Elapsed: 81ms
Info 319  [12:8:10.630] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 320  [12:8:11.426] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 321  [12:8:11.492] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 58 structureChanged: false Elapsed: 66ms
Info 322  [12:8:11.492] Different program with same set of files:: oldProgram.structureIsReused:: 2
Info 323  [12:8:12.497] Starting updateGraphWorker: Project: /home/my/project/tsconfig.json
Info 324  [12:8:12.562] Finishing updateGraphWorker: Project: /home/my/project/tsconfig.json Version: 59 structureChanged: false Elapsed: 65ms
Info 325  [12:8:12.562] Different program with same set of files:: oldProgram.structureIsReused:: 2
```

**Comment by ayazhafiz** [comment]
Okay, nothing unexpected here. What about right before it breaks? Does it look the same?

**Comment by tonivj5** [comment]
> What about right before it breaks? Does it look the same?

I don't understand exactly what you mean.

Right before it breaks, vscode doesn't give any intellisense in the template. When I move to another file, it shows the errors I commented here (https://github.com/angular/vscode-ng-language-service/issues/713#issuecomment-612593758). If I write (for example) a blankspace, the errors go away and the intellisense in every template is broken (no intellisense).

It simply, stop working :disappointed:

I have learn how to break it more or less reproduciblement. I have done a gif:
![error-vscode-ng](https://user-images.githubusercontent.com/7110786/79471392-d92b3980-8002-11ea-9777-9066f1f48512.gif)

It seems that if you edit, broke and repair the template in a few seconds. It breaks the plugin.

**Comment by ayazhafiz** [comment]
I mean what does the log file say right before this breaks?

My best guess is that the language service is taking far too long to respond and requests get cancelled.

**Comment by tonivj5** [comment]
Yeah, it looks the same. When the plugin breaks. The log doesn't add any line more, it gets stuck there.

**Comment by ayazhafiz** [comment]
Okay, I see.

________________________________
От: Toni Villena <notifications@github.com>
Отправлено: Thursday, April 16, 2020 1:26:11 PM
Кому: angular/vscode-ng-language-service <vscode-ng-language-service@noreply.github.com>
Копия: hafiz <ayaz.hafiz.1@gmail.com>; Mention <mention@noreply.github.com>
Тема: Re: [angular/vscode-ng-language-service] Angular Language Service 0.901 not activating on VS Code 1.43.2 (#713)


Yeah, it looks the same. When the plugin breaks. The log doesn't add any line more, it gets stuck there.

**Comment by ayazhafiz** [comment]
Yeah sorry I haven’t been super helpful. I’ve been bogged with work and illness so haven’t found a couple hours to sit down and figure this out. When I get a chance I’ll try to sit down with a big angular app and figure out what’s up.

If you notice any patterns in when this happens feel free to let them known in this chat. It sounds like this wasn’t a problem before this version, so reverting may help with the issues.

**Comment by tonivj5** [comment]
Okay! Thank you for you effort :100:

**Comment by kyliau** [comment]
@tonivj5 Could you please take a screenshot of the *output* panel?
It should look something like this:
![Screen Shot 2020-04-16 at 5 02 02 PM](https://user-images.githubusercontent.com/2941178/79517800-1cd27300-8004-11ea-8be0-353a644c0bf7.png)

**Comment by tonivj5** [comment]
@kyliau, sure! Here you have :+1: 
![vscode-ng-template](https://user-images.githubusercontent.com/7110786/79519902-85960b80-8055-11ea-8169-b77e634a8102.png)

**Comment by kyliau** [comment]
The problem here seems like after some period of editing, the language service stops responding to user changes and gets stuck in a bad/stale state. This is not a project loading problem, as alluded to in the title. I'll spend some time to investigate, but as you've shown, getting a reproduction is not exactly straight-forward. In the meantime, as a _temporary_ workaround, I suggest **restarting the language service**, as shown in the screenshot below.

![Screen Shot 2020-04-16 at 7 34 40 PM](https://user-images.githubusercontent.com/2941178/79526147-7abd8580-8019-11ea-9c76-6e2333ca38af.png)

**Comment by alexermolov** [comment]
I have an update with my crash.

![pic](https://user-images.githubusercontent.com/25246926/79558193-d5f08600-80b4-11ea-9eca-d3a8c6cb0b7d.png)

**Comment by tonivj5** [comment]
@kyliau and @ayazhafiz, I have been able of reproduce in a [`ng new` project that I have to report issues](https://github.com/tonivj5/angular-issues-35424-and-35426)

Here are the steps (it takes me a few minutes):
- Open the project
- Open the files inside `language-service-breaks` folder and `app.component.html`
- Do some work hovering the mouse over `list-box.component.ts`' template and pressing `ctrl` + mouse hover.
- Go to `app.component.html`, do some work in that template, all is ok.
- Go to `list.component.ts` (you can break any other file but I'm using this :laughing:) and start editing it: break the template and repair it, break, save, repair, break, do some intellisense, repair, save, break, do some intellisense, break, ... it's very tedious :sweat:
- Sometime, it's going to break. When it happens, if you go to `list-box.component.ts`, it works. But, if you edit again the `list.component.ts` and go back to `list-box.component.ts`, it was broken, and all templates of `*.ts` files were broken too (inline templates).
- Go to `app.component.html`, only `*ngIf` is broken (when it was working before), it happens with other `common` directives (`*ngFor`, `*ngSwitch`, etc.) but not with my directives. But, you can break them too, doing the same steps (break, repair, save, intellisense, .....). It breaks all `*.html` files.

I have done a (a bit long, 3 minutes) gif to show you
![ng-breaks](https://user-images.githubusercontent.com/7110786/79560124-42fd1f00-80a7-11ea-8432-ff80f0abeeab.gif)

Using a bigger project, it's really easy to break it. Sometimes it take longer or shorter time to reproduce it, but at the end, it always breaks. And I have noticed, that inline and file templates doesn't broke at the same time (at least, not totally).

**Comment by ivanwonder** [comment]
@tonivj5 The way to reproduce, first press ctrl + mouse hove over `emit()`, then press ctrl + mouse hover over `*ngIf`.
Or another way to reproduce that is open file `core.d.ts`, then open file `common.d.ts`.

When the user opens the file `core.d.ts`, the content of `core.d.ts` is different from content in the disk, the angular language service will remove the symbols from `core.d.ts`, I think some symbols were created when the compiler initializes.
@kyliau @ayazhafiz Maybe we should ignore the `core.d.ts` when updating module.
https://github.com/angular/angular/blob/master/packages/language-service/src/typescript_host.ts#L245-L248

Looks like some of the end of the line in `core.d.ts` are `CR`, and some are `CRLF`. when vscode opens it, it will be formatted. 
https://github.com/angular/vscode-ng-language-service/blob/4236352238aaa8355f2c74ba19999622bc790240/server/src/session.ts#L177
`text !== ts.sys.readFile(filePath)`

@tonivj5 So if you edit the `core.d.ts` and save it, it should resolve the problem temporarily.

**Comment by tonivj5** [comment]
> @tonivj5 The way to reproduce, first press ctrl + mouse hove over emit(), then press ctrl + mouse hover over *ngIf.
Or another way to reproduce that is open file core.d.ts, then open file common.d.ts.

OMG, good catch!! :open_mouth: I confirm it breaks :+1:

> @tonivj5 So if you edit the core.d.ts and save it, it should resolve the problem temporarily.

Hmm, it doesn't resolve the problem to me... I don't know if I'm doing something wrong. If I open `core.d.ts` and edit it, it breaks completely :sob: I'm using Linux :penguin:, if it helps.

Thanks for test it and find where the bug comes :clap::clap::clap:

**Comment by ivanwonder** [comment]
@tonivj5 after you save the file,you need reload the vscode.

**Comment by tonivj5** [comment]
Okei, I have copy `core.d.ts` into my `language-service-breaks` folder, I have edit it and now, it has been formated (using prettier). I have copy the content to `core.d.ts` of `node_modules` and now I can hover over `*ngIf` and `emit`, it isn't break :tada:

You're a magician @ivanwonder :mage_man:

**Comment by ivanwonder** [comment]
@tonivj5 I think no need for `common.d.ts`. To be exact,just make sure the end of line of `core.d.ts` are all the same.

**Comment by tonivj5** [comment]
Surely! It must be my fault, but editing both files, it worked

**Comment by kyliau** [comment]
> When the user opens the file `core.d.ts`, the content of `core.d.ts` is different from content in the disk, the angular language service will remove the symbols from `core.d.ts`, I think some symbols were created when the compiler initializes.
> @kyliau @ayazhafiz Maybe we should ignore the `core.d.ts` when updating module.

Yes, it is indeed true that when the Angular language service is initialized, it loads core symbols from `core.d.ts` in the compiler.

> Looks like some of the end of the line in `core.d.ts` are `CR`, and some are `CRLF`. when vscode opens it, it will be formatted.

We have not made any changes to this part of the codebase in v0.901.0, but one PR that could potentially affect this behavior is https://github.com/angular/angular/pull/35262.

**Comment by ivanwonder** [comment]
I find that the end of the line of license banner is LF, and the other generated code is CRLF.

**Comment by ivanwonder** [comment]
@kyliau I have submitted a PR, I think we also can prompt the user to restart the Angular language service.

**Comment by tonivj5** [comment]
> @tonivj5 The way to reproduce, first press ctrl + mouse hove over emit(), then press ctrl + mouse hover over *ngIf.

I'm trying it in the latest version (`0.901.4`) and it seems it doesn't break :tada:

Thanks you all :clap:

**Comment by kyliau** [comment]
Thank you @ivanwonder for narrowing down the issue and submitting a fix! I'll close this issue for now. Please open a **new issue** if similar problem is encountered.
